composer/callbacks/activation_monitor.py

```diff
@@ -0,0 +1,216 @@
+    torch module. This is done by attaching a forward hook to the module. Additionally, when after we finish logging
+    we detach the forwards hook.
+
+    Example:
+        .. doctest::
+
+            >>> from composer import Trainer
+            >>> from composer.callbacks import ActivationMonitor
+            >>> # constructing trainer object with this callback
+            >>> trainer = Trainer(
+            ...     model=model,
+            ...     train_dataloader=train_dataloader,
+            ...     eval_dataloader=eval_dataloader,
+            ...     optimizers=optimizer,
+            ...     max_duration="1ep",
+            ...     callbacks=[ActivationMonitor()],
+            ... )
+
+    The metrics are logged by the :class:`.Logger` to the following keys described below. For convenience we have included
+    example metrics logged below:
+
+        +-------------------------------------------------------+-----------------------------------------------------+
+        | Key                                                   | Logged data                                         |
+        +=======================================================+=====================================================+
+        |                                                       | The average max value of the last dimension of the  |
+        | ``activations/max/MODULE_NAME/input_{n}``             | nth input activations into the current module.      |
+        |                                                       |                                                     |
+        +-------------------------------------------------------+-----------------------------------------------------+
+        |                                                       | The average value of the nth input activations      |
+        | ``activations/average/MODULE_NAME/input_{n}``         | into the current module.                            |
+        |                                                       |                                                     |
+        +-------------------------------------------------------+-----------------------------------------------------+
+        |                                                       | The average L2 Norm of the last dimension of the    |
+        | ``activations/l2_norm/MODULE_NAME/input_{n}``         | nth input activations into the current module.      |
+        |                                                       |                                                     |
+        +-------------------------------------------------------+-----------------------------------------------------+
+        |                                                       | The kurtosis of the last dimension of the nth input |
+        | ``activations/kurtosis/MODULE_NAME/input_{n}``        | activations into the current module.                |
```

Comment:
I think we want the kurtosis measured across the model dimension, so I'm a bit confused about your wording here. Can you clarify by showing for a tensor of size (B, seq_len, d_model) what you are calculating?

For instance, I think we should first average over B and seq_len to get vectors of size d_model, then compute our statistics over those averaged d_model sized vectors. Let me know if that agrees with your intuition and lets find a way to be extra careful that that's what we are actually doing.

Reply:
Actually I'm not even sure.. I believe the below graph suggests we should compute statistics over the dimension of d_model, then average over B and seq_len? Let me know if that makes sense to you. At any rate we should have one number per layer.

<img width="882" alt="image" src="https://user-images.githubusercontent.com/115047575/226693720-04e82982-1020-4bdb-9af9-8f773f544dc7.png">

Reply:
Yeah I'm doing statistics over d_model. Happy to clarify